### DISCUSSION TITLE
In the useMutation function, why is the mutate type for TVariables FormData & void?

### DISCUSSION THREAD

**Discussion by gyong0508** [TanStack/query#6033] [Q&A]
![image](https://github.com/TanStack/query/assets/102498338/f319b4e5-7aa4-4281-9f7e-5a61a19fc559)
![image](https://github.com/TanStack/query/assets/102498338/6a30f3a3-c59c-41e6-bdaf-7507d19804d1)
![image](https://github.com/TanStack/query/assets/102498338/0ef1fe60-3bf6-4bb0-9773-e522c1901243)

**Comment by TkDodo** [comment]
Show a typescript playground reproduction please

**Reply by gyong0508** [reply]
"Don't mind this; the problem is in my code. The issue is that the `useUpdateDocuments` hook has two sources. The first source doesn't have a type definition, only the second one does. TypeScript combined the types of the two sources, which is why it includes `void`."
